### DISCUSSION TITLE
GitVersion and Fast-forward Merge - Can this work?

### DISCUSSION THREAD

**Discussion by ChristosMonogios** [GitTools/GitVersion#3671] [Q&A] [answered]
Hello all

I am setting up GitVersion for some repositories of mine.
I am using mode: Mainline and push my merge requests to main, When my repositories are using "Merge commit" as merge method then GitVersion is able to match the regex I defined in GitVersion.yml and, for example, create a minor version from a branch with the string "task" in it.

However, when I am not using the "Merge commit" option but instead I use the "Fast-forward merge" option, no merge commit is created and I guess GitVersion cannot figure out from which feature branch the change came.

My question is, can "Fast-forward merge" and GitVersion play along? Or I can only use "Merge commit" and create new commit messages every time something is merged into main?

**Comment by HHobeck** [comment, marked as answer]
> However, when I am not using the "Merge commit" option but instead I use the "Fast-forward merge" option, no merge commit is created and I guess GitVersion cannot figure out from which feature branch the change came.

Yes I can confirm your analysis.

> My question is, can "Fast-forward merge" and GitVersion play along? Or I can only use "Merge commit" and create new commit messages every time something is merged into main?

If the merge commit message is important to determine the version number then yes you should not merge with fast-forward mode. In your scenario it is important because git-version needs the information from where the commits are coming from. As an alternative you can use bump message like `+semver: minor` or use a different release and branching strategy.
